cloud/amazon/ec2_customer_gateway.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_customer_gateway
+short_description: Manage an AWS customer gateway
+description:
+    - Manage an AWS customer gateway
+version_added: "2.1"
+author: Michael Baydoun (@MichaelBaydoun)
+requirements: [ boto3 ]
+options:
+  bgp_asn:
+    description:
+      - Border Gateway Protocol (BGP) Autonomous System Number (ASN), required when state=present
+    required: false
+    default: null
+  ip_address:
+    description:
+      - Internet-routable IP address for customers gateway, must be a static address
+    required: true
+  name:
+    description:
+      - Name of the customer gateway
+    required: true 
+  state:
+    description:
+      - Create or terminate the Customer Gateway
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+- ec2_customer_gateway:
+    bgp_asn: 12345
+    ip_address: 1.2.3.4
+    region: us-east-1
+  register: cgw
+
+- ec2_customer_gateway:
+    ip_address: 1.2.3.4
+    state: absent
+    region: us-east-1
+  register: cgw
+'''
+
+RETURN = '''
+gateway.CustomerGateway:
+    description: details about the gateway that was created
+    returned: success
+    type: complex
+    contains:
+        BgpAsn:
+            description: The Border Gateway Autonomous System Number
+            returned: when exists and gateway is available
+            sample: 65123
+            type: string
+        CustomerGatewayId:
+            description: gateway id assigned by amazon 
+            returned: when exists and gateway is available
+            sample: cgw-cb6386a2
+            type: string
+        IpAddress:
+            description: ip address of your gateway device 
+            returned: when exists and gateway is available
+            sample: 1.2.3.4
+            type: string
+        State:
+            description: state of gateway 
+            returned: when gateway exists and is available 
+            state: available
+            type: string
+        Type:
+            description: encryption type 
+            returned: when gateway exists and is available 
+            sample: ipsec.1 
+            type: string
+'''
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
```

Comment:
Half way there, the module will currently fail with a backtrace if boto3 isn't installed. This isn't clear in the Developing Module guide. I'll get that updated.

You to check this as well, something similar to

```
if not HAS_BOTO3:                                                        
         module.fail_json(msg='boto3 is required for this module')
```